agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/exporter/AzureMonitorMetricExporter.java

```diff
@@ -0,0 +1,112 @@
+import com.azure.monitor.opentelemetry.exporter.implementation.models.TelemetryItem;
+import com.azure.monitor.opentelemetry.exporter.implementation.utils.FormattedTime;
+import com.microsoft.applicationinsights.agent.internal.telemetry.TelemetryClient;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.data.MetricDataType;
+import io.opentelemetry.sdk.metrics.data.PointData;
+import io.opentelemetry.sdk.metrics.export.MetricExporter;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AzureMonitorMetricExporter implements MetricExporter {
+
+  private static final List<String> EXCLUDED_METRIC_NAMES = new ArrayList<>();
+
+  private final TelemetryClient telemetryClient;
+  private static final Logger logger = LoggerFactory.getLogger(AzureMonitorMetricExporter.class);
+  private final AtomicBoolean stopped = new AtomicBoolean();
+
+  public AzureMonitorMetricExporter(TelemetryClient telemetryClient) {
+    this.telemetryClient = telemetryClient;
+  }
+
+  static {
+    EXCLUDED_METRIC_NAMES.add("http.server.active_requests"); // Micrometer
+    EXCLUDED_METRIC_NAMES.add("http.server.duration"); // Micrometer
+    EXCLUDED_METRIC_NAMES.add("http.client.duration"); // HttpClient
+    EXCLUDED_METRIC_NAMES.add("rpc.client.duration"); // gRPC
+    EXCLUDED_METRIC_NAMES.add("rpc.server.duration"); // gRPC
+  }
+
+  @Override
+  public CompletableResultCode export(Collection<MetricData> metrics) {
+    if (stopped.get()) {
+      return CompletableResultCode.ofFailure();
+    }
+
+    for (MetricData metricData : metrics) {
+      if (EXCLUDED_METRIC_NAMES.contains(metricData.getName())) {
+        continue;
+      }
+
+      MetricDataType type = metricData.getType();
+      if (type == DOUBLE_SUM
+          || type == DOUBLE_GAUGE
+          || type == LONG_SUM
+          || type == LONG_GAUGE
+          || type == HISTOGRAM) {
+        convertOtelMetricToAzureMonitorMetric(metricData).forEach(telemetryClient::trackAsync);
+      } else {
+        logger.warn("metric data type {} is not supported yet.", type);
+      }
+    }
+    return CompletableResultCode.ofSuccess();
+  }
+
+  @Override
+  public CompletableResultCode flush() {
+    return CompletableResultCode.ofSuccess();
+  }
+
+  @Override
+  public CompletableResultCode shutdown() {
+    stopped.set(true);
+    return CompletableResultCode.ofSuccess();
+  }
+
+  private List<TelemetryItem> convertOtelMetricToAzureMonitorMetric(MetricData metricData) {
+    List<TelemetryItem> telemetryItems = new ArrayList<>();
+    for (PointData data : metricData.getData().getPoints()) {
+      MetricTelemetryBuilder builder = telemetryClient.newMetricTelemetryBuilder();
+      builder.setTime(FormattedTime.offSetDateTimeFromNow());
```

Comment:
I believe otel metric data has its own time, let's use that instead of "now"

Reply:
i didn't change it.. it will use whatever otel default. .net is the same. should we get some consensus from other platform first?

Reply:
which one:

 ```
datapoint:
this.startEpochNanos
this.epochNanos
```

i think use now makes more sense since that's when azure monitor metric is created.  in my opinion the time in otel metric doesn't matter.

Reply:
I believe the "correct" time is the time that the metric was gathered (e.g. for cpu metric it would be what time the cpu metric was read from the O/S), so in this sense I think the otel metric time is better